core/modules/replica/ConfigurationTypes.h

```diff
@@ -0,0 +1,418 @@
+/*
+ * LSST Data Management System
+ * Copyright 2019 LSST Corporation.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_REPLICA_CONFIGURATIONTYPES_H
+#define LSST_QSERV_REPLICA_CONFIGURATIONTYPES_H
+
+/**
+ * This header defines ancillary helper types which are meant to reduce code
+ * duplications in applications dealing with class Configuration.
+ */
+
+// System headers
+#include <string>
+
+// Third party headers
+#include "nlohmann/json.hpp"
+
+// Qserv headers
+#include "replica/Configuration.h"
+
+// This header declarations
+
+namespace lsst {
+namespace qserv {
+namespace replica {
+
+/**
+ * Structure ConfigurationGeneralParams defines helpers for the general parameters
+ * of the Configuration system's API.
+ */
+struct ConfigurationGeneralParams {
+
+    struct {
+
+        std::string const key         = "NET_BUF_SIZE_BYTES";
+        std::string const description = "The default buffer size for network communications.";
+        size_t            value;
+
+        bool const updatable = true;
```

Comment:
It's not clear to me (yet, anyway) how `updatable` is used. It seems that these are meant to be simple, lightweight, helper classes, but I wonder if some documentation is appropriate. Maybe you could explain the pattern a little more in the comment around line 45? If you gave a summary of the expected parameters & functions of each struct are and how it is to be used I think it would be helpful.

Reply:
These attributes are used by the CLI and Web UI tools to determine if the Configuration parameter is allowed to be updated before even attempting it ... failing. For the Web UI these parameters will be "grayed out" in the UI. In the CLI tools (`qserv-replica-config`) they won't be allowed to be modified.
I could put comments explaining these structures in the header file. This was a good suggestion.

Reply:
I think it would be helpful to have those comments.

Reply:
Comments added.